androidApp/src/main/java/com/example/tasky/android/alarm/data/AlarmReceiver.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.tasky.android.alarm.data
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import com.example.tasky.agenda.domain.model.AgendaType
+
+class AlarmReceiver : BroadcastReceiver() {
+    override fun onReceive(
+        context: Context?,
+        intent: Intent?,
+    ) {
+        val id = intent?.getStringExtra(AlarmIntentKey.ID.name) ?: return
+        val title = intent.getStringExtra(AlarmIntentKey.TITLE.name) ?: return
+        val type =
```

Comment:
Considering that you have multiple pieces of data that you need in order to create a new alarm, what about communicating that in the code via a data class? With that approach, you have a single `null` check, and then you can freely access all the properties. I’m thinking of something like this:

```
// this is a data class to separate the @Parcelable annotation from the domain class: NotificationData
@Parcelize
data class NotificationDataParcelable(
    val id: String,
    val title: String,
    val description: String?, // <— nullable because the description is optional
    val notifyAtMillis: Long
) : Parcelable
```